frontend/eda/event-streams/hooks/useEventStreamActions.tsx

```diff
@@ -0,0 +1,91 @@
+  const { t } = useTranslation();
+  const disableEventStreams = useDisableEventStreams(view.unselectItemsAndRefresh);
+  const restartEventStreams = useRestartEventStreams(view.unselectItemsAndRefresh);
+  const deleteEventStreams = useDeleteEventStreams(view.unselectItemsAndRefresh);
+  const alertToaster = usePageAlertToaster();
+  const enableEventStream: (eventStream: EdaEventStream) => Promise<void> = useCallback(
+    async (eventStream) => {
+      const alert: AlertProps = {
+        variant: 'success',
+        title: `${eventStream.name} ${t('enabled')}.`,
+        timeout: 5000,
+      };
+      await postRequest(edaAPI`/event-streams/${eventStream.id.toString()}/${'enable/'}`, undefined)
+        .then(() => alertToaster.addAlert(alert))
+        .catch(() => {
+          alertToaster.addAlert({
+            variant: 'danger',
+            title: `${t('Failed to enable')} ${eventStream.name}`,
+            timeout: 5000,
+          });
+        });
+      view.unselectItemsAndRefresh([eventStream]);
+    },
+    [view, alertToaster, t]
+  );
+
+  return useMemo<IPageAction<EdaEventStream>[]>(() => {
+    const actions: IPageAction<EdaEventStream>[] = [
+      {
+        type: PageActionType.Switch,
+        ariaLabel: (isEnabled) =>
+          isEnabled ? t('Click to disable instance') : t('Click to enable instance'),
+        selection: PageActionSelection.Single,
+        isPinned: true,
+        label: t('Event stream enabled'),
+        labelOff: t('Event stream disabled'),
+        onToggle: (eventStream: EdaEventStream, activate: boolean) => {
+          if (activate) void enableEventStream(eventStream);
+          else void disableEventStreams([eventStream]);
+        },
+        isSwitchOn: (eventStream: EdaEventStream) => eventStream.is_enabled ?? false,
+        isHidden: (eventStream: EdaEventStream) => eventStream?.status === Status906Enum.Deleting,
+        isDisabled: (eventStream: EdaEventStream) =>
+          eventStream.status === Status906Enum.Stopping
+            ? t('Cannot change the event stream status while stopping')
+            : undefined,
+      },
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Single,
+        icon: RedoIcon,
+        label: t('Restart event stream'),
+        isHidden: (eventStream: EdaEventStream) =>
+          !eventStream.is_enabled || eventStream?.status === Status906Enum.Deleting,
+        onClick: (eventStream: EdaEventStream) => restartEventStreams([eventStream]),
+      },
+      {
+        type: PageActionType.Seperator,
+      },
+      {
+        type: PageActionType.Button,
+        selection: PageActionSelection.Single,
+        icon: TrashIcon,
+        label: t('Delete event stream'),
+        isHidden: (eventStream: EdaEventStream) => eventStream?.status === Status906Enum.Deleting,
+        onClick: (eventStream: EdaEventStream) => deleteEventStreams([eventStream]),
+        isDanger: true,
+      },
+    ];
+    return actions;
```

Comment:
can we have data-cys added to the above elements here?

Reply:
the component used here already adds a data-cy for each menu item

Reply:
data-cy added by component